portfolio/src/main/webapp/greeting.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * Uses arrow functions to fetch and display greeting from the data servlet.
+ */
+function getRandomGreeting() {
+  fetch('/data').then(response => response.text()).then((quote) => {
+    document.getElementById('greeting').innerText = quote;
```

Comment:
Isn't this going to print a JSON object?  Why convert it to text?  Just work with it as json to get what you want.

Reply:
I have removed the conversion and changed the name of the function to better reflect its purpose.